### PR TITLE
Network interface removal and decoupling

### PR DESCRIPTION
**Pull request by sameo**
This PR does 2 things:

- Remove the network interface (See #1179)
- Decouple the network API from the virtcontainers one (See #1180)

### CODE REVIEW

**Comment by sameo** [comment]
> @sameo I'm not against the removal of the Go interface since, as you mentioned it, we're only using one single implementation. But I'm concerned about the unit tests, because the noop implementation is pretty convenient to test a bunch of code that would be complex to test if we had to mimic the real network behavior. WDYT?

FWIW all the unit tests are passing with the default network implementation. When first thinking about doing this I thought they will all fail but they don't. I get CI errors here but there not UT related.

File: virtcontainers/network.go [hunk 1]
```diff
@@ -1464,17 +1417,91 @@ func createEndpoint(netInfo NetworkInfo, idx int, model NetInterworkingModel) (E
...
-// Container network plugins are used to setup virtual network
-// between VM netns and the host network physical interface.
-type network interface {
-	// run runs a callback function in a specified network namespace.
-	run(networkNSPath string, cb func() error) error
+// Network is the virtcontainer network structure
+type Network struct {
+}
+
+func (n *Network) trace(ctx context.Context, name string) (opentracing.Span, context.Context) {
```
Comment (mcastelino) [inline virtcontainers/network.go]: @sameo This is code moved mostly unmodified from virtcontainers/default_network.go?

Where there any changes needed?

File: virtcontainers/sandbox.go [hunk 2]
```diff
@@ -206,7 +205,7 @@ type Sandbox struct {
 	hypervisor hypervisor
 	agent      agent
 	storage    resourceStorage
-	network    network
+	network    Network
```
Comment (mcastelino) [inline virtcontainers/sandbox.go]: @sameo why did this need to be made public?

**Comment by sameo** [comment]
@sboeuf @mcastelino https://github.com/kata-containers/runtime/pull/1181/commits/b39cb1d13a283322a489f9a746eeb8f99e173eda#diff-bca2c2b5b04e0e74533b492fd23a3cc8R784 is what makes several unit tests pass with this PR.

File: virtcontainers/sandbox.go (hunk 2 above)
Comment (sameo) [reply virtcontainers/sandbox.go]: I'm preparing for moving the network code into its own package.

File: virtcontainers/network.go (hunk 1 above)
Comment (sameo) [reply virtcontainers/network.go]: No functional changes, no.

**Comment by amshinde** [comment]
@sameo Change looks good overall, its just the noop implementation has been useful in mocking our network behaviour in unit tests.

**Comment by sameo** [comment]
> @sameo Change looks good overall, its just the noop implementation has been useful in mocking our network behaviour in unit tests.

noop is useful for sure, but:

* It does not exercise much of the code path
* We need to maintain an artificial abstraction for it
* All unit tests pass without it

**Comment by amshinde** [comment]
@sameo If thats the case
lgtm

**Comment by bergwolf** [comment]
LGTM! It makes sense to simplify the code and `CNM` is not really CNM specific anyway.

**Comment by sboeuf** [comment]
@chavafg which of those CI results can be ignored?

**Comment by GabyCT** [comment]
@sboeuf here it is the page where it says the `CI known failures` http://jenkins.katacontainers.io/view/CI%20Status/, what I see that is missing is the `metrics` job. @grahamwhaley could you please add that with the proper link to the issue. Thanks

**Comment by sboeuf** [comment]
thanks @GabyCT 
I've just restarted the fedora and metrics builds, let see how this goes.

**Pull request merged**